portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -34,23 +34,41 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);;
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
     ArrayList<String> commentSection = new ArrayList<String>();
     for (Entity entity : results.asIterable()) {
         String text = (String) entity.getProperty("text");
+        commentSection.add(text);
+    }
 
-        String comment = text;
-        commentSection.add(comment);
+    int commentCount = getRequestedCommentCount(request);
+    
+    ArrayList<String> limitedComments = new ArrayList<String>();
+    for (int i = 0; i < commentCount; i++) {
+        limitedComments.add(commentSection.get(i));
     }
```

Comment:
You don't need to do this work here, you can instead add the number of results you want directly in the query you have above using SELECT TOP, LIMIT or equivalent